### PR TITLE
[PLUGIN-1209] Failed to deploy the plugin using CloudSQL MySQL & Postgres source and sink with connection

### PR DESCRIPTION
**Pull request by itsankit-google**
Fixes - [PLUGIN-1209](https://cdap.atlassian.net/browse/PLUGIN-1209)

### CODE REVIEW

File: cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLSink.java [hunk 1]
```diff
@@ -54,11 +54,14 @@ public CloudSQLMySQLSink(CloudSQLMySQLSinkConfig cloudsqlMysqlSinkConfig) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
-    
-    CloudSQLMySQLUtil.checkConnectionName(
+
+    if (!cloudsqlMysqlSinkConfig.containsMacro(CloudSQLMySQLConstants.INSTANCE_TYPE) &
```
Comment (yaojiefeng) [inline cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLSink.java]: `&` -> `&&`

File: cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLSource.java [hunk 2]
```diff
@@ -57,10 +57,13 @@ public CloudSQLMySQLSource(CloudSQLMySQLSourceConfig cloudsqlMysqlSourceConfig)
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
 
-    CloudSQLMySQLUtil.checkConnectionName(
+    if (!cloudsqlMysqlSourceConfig.containsMacro(CloudSQLMySQLConstants.INSTANCE_TYPE) &
```
Comment (yaojiefeng) [inline cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLSource.java]: `&` -> `&&`

File: cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLUtil.java [hunk 3]
```diff
@@ -33,27 +35,45 @@ public class CloudSQLMySQLUtil {
    * @param connectionName Connection Name for the CloudSQL MySQL instance
    */
   public static void checkConnectionName(
-      FailureCollector failureCollector, String instanceType, String connectionName) {
+    FailureCollector failureCollector, @Nullable String instanceType, @Nullable String connectionName) {
+
+    if (Strings.isNullOrEmpty(instanceType)) {
```
Comment (yaojiefeng) [inline cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLUtil.java]: So if this cannot be null or empty, can we mark these as required fields?

File: cloudsql-postgresql-plugin/src/main/java/io/cdap/plugin/cloudsql/postgres/CloudSQLPostgreSQLUtil.java [hunk 4]
```diff
@@ -35,27 +37,45 @@ public class CloudSQLPostgreSQLUtil {
    * @param connectionName Connection Name for the CloudSQL PostgreSQL instance
    */
   public static void checkConnectionName(
-      FailureCollector failureCollector, String instanceType, String connectionName) {
-    
+    FailureCollector failureCollector, @Nullable String instanceType, @Nullable String connectionName) {
+
+    if (Strings.isNullOrEmpty(instanceType)) {
```
Comment (yaojiefeng) [inline cloudsql-postgresql-plugin/src/main/java/io/cdap/plugin/cloudsql/postgres/CloudSQLPostgreSQLUtil.java]: Same here, and this looks pretty much same as the mysql ones, better combine them

File: cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLUtil.java (hunk 3 above)
Comment (itsankit-google) [reply cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLUtil.java]: ack

File: cloudsql-postgresql-plugin/src/main/java/io/cdap/plugin/cloudsql/postgres/CloudSQLPostgreSQLUtil.java (hunk 4 above)
Comment (itsankit-google) [reply cloudsql-postgresql-plugin/src/main/java/io/cdap/plugin/cloudsql/postgres/CloudSQLPostgreSQLUtil.java]: ack

File: cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLSource.java (hunk 2 above)
Comment (itsankit-google) [reply cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLSource.java]: oops!

File: cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLSink.java (hunk 1 above)
Comment (itsankit-google) [reply cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLSink.java]: oops!

File: database-commons/src/main/java/io/cdap/plugin/util/CloudSQLUtil.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+   * Utility method to check the Connection Name format of a CloudSQL instance.
+   *
+   * @param failureCollector {@link FailureCollector} for the pipeline
+   * @param instanceType CloudSQL instance type
+   * @param connectionName Connection Name for the CloudSQL instance
+   */
+  public static void checkConnectionName(
+    FailureCollector failureCollector, @Nullable String instanceType, @Nullable String connectionName) {
+
+    if (Strings.isNullOrEmpty(instanceType)) {
```
Comment (yaojiefeng) [inline database-commons/src/main/java/io/cdap/plugin/util/CloudSQLUtil.java]: If these fields are marked as required in the config, we can just remove the null check here, as the plugin won't be able to get instantiated, considering you already do the macro check before calling this method

**Review by yaojiefeng — APPROVED** [review]
One comment, LGTM

File: database-commons/src/main/java/io/cdap/plugin/util/CloudSQLUtil.java (hunk 5 above)
Comment (itsankit-google) [reply database-commons/src/main/java/io/cdap/plugin/util/CloudSQLUtil.java]: I see, sg.

**Pull request merged**